packages/ra-core/src/form/FormWithRedirect.tsx

```diff
@@ -103,9 +104,25 @@ const FormWithRedirect = ({
         [save]
     );
 
+    const subscribers = useRef<{
+        [key: string]: FormGroupSubscriber[];
+    }>({});
+
     const formContextValue = useMemo<FormContextValue>(
         () => ({
             setOnSave,
+            subscribe: (group, subscriber) => {
```

Comment:
Can you please explain what you're doing in a comment?

Reply:
I will extract all this inside a custom hook in another PR